libmobilecoin/src/chacha20_rng.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use crate::common::*;
+use mc_util_ffi::*;
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+use std::{convert::TryInto, sync::Mutex};
+
+pub type McChaCha20Rng = ChaCha20Rng;
+
+// McU128 facilitates conversion between u128 and 16 bytes of u8
+pub struct McU128 {
+    pub bytes: [u8; 16],
+}
+
+impl McU128 {
+    pub fn from_u128(val: u128) -> McU128 {
```

Comment:
Consider adding `new`/`from_bytes`

```suggestion
    pub fn new(bytes: [u8; 16]) -> McU128 {
        McU128 { bytes }
    }

    pub fn from_u128(val: u128) -> McU128 {
```

Reply:
McU128 isn't really necessary - removed it after making the change below from one of James' review comments removing the dependency.